images/kamu-base-git/README.md

```diff
@@ -0,0 +1,36 @@
+# Building `kamu` in Docker
+
+This can be useful if you need a `kamu-cli` build, but don't have the desire/ability 
+to roll out a build environment and all the tools.
+
+⚠️ Please note: for flexibility, this image does NOT include any kamu workspace.
+
+### Build
+
+```shell
+cd ./images/kamu-base-git/
+docker build -t kamu-base:git .
+```
+
+Or, you can also specify the desired branch:
+```shell
+docker build --build-arg KAMU_BRANCH=<REPLACE_ME> -t kamu-base:git .
+```
```

Comment:
What we discussed with hauki team last week is this vision:
- every repo has a "dev build" GH action
  - runs on demand
  - parametrized by branch name / commit
  - generates deployment artifacts (docker images, helm charts)
- artifacts are stored in a dev image registry
  - separately from production images
  - a place we can easily clean up when needed
- a "Dev Deployment" GH action will exist e.g. in `kamu-deploy` repo
  - is parametrized with branch names per every component
  - it kicks off "dev build" actions in all repos
  - waits for completion
  - then proceeds to deploy the artifacts to dev environment

If we follow this plan - I think there still exists a need for "dev docker image", but there is no need to build kamu in docker.

I assume you are creating this image mainly for @dmitriy-borzenko to be able to test UI against latest branch changes, but I think it would be a lot more convenient for him to just trigger such image builds on GitHub instead of building locally:
- it's just easier to use
- there's less risk of human error - forgetting to pull, building wrong commit etc
- GH job can reuse cache and given how resource-hungry our rust builds are just not stress frontend dev's laptops

Reply:
> If we follow this plan - I think there still exists a need for "dev docker image", but there is no need to build kamu in docker.

I really like this plan, it's the way it's supposed to be.

> I assume you are creating this image mainly for @dmitriy-borzenko to be able to test UI against latest branch changes, but I think it would be a lot more convenient for him to just trigger such image builds on GitHub instead of building locally:

This PR is not offering an alternative, but a solution to the here and now -- since Dmitriy couldn't work with any branch of `kamu-cli` as we previously wanted. 

I'll be happy to remove this image as soon as the above GH actions are available, but for now I suggest leaving it as is, as it has already proven to be effective.

Reply:
Ok let's solve the immediate problem, but later on I see this image likely reused (not replaced) in the "dev build" GH action.

@dmitriy-borzenko let us know if you find this build procedure too cumbersome - creating GH action to automate it should be a simple matter.

Reply:
At the moment, Dima's solution solves the problem. But if it can be simplified and automated, I will be only glad